file.go

```diff
@@ -1,37 +1,176 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	data []map[string]interface{}
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	// open JSON file
+	jsonFile, err := os.Open(path)
+
+	if err != nil {
+		return file{}, err
+	}
+
+	defer jsonFile.Close()
+
+	return ParseOpenedJsonFile(jsonFile)
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	// read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(f)
+
+	if err != nil {
+		return file{}, err
+	}
+
+	return ParseJson(byteValue)
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	// open CSV file
+	csvFile, err := os.Open(path)
+
+	if err != nil {
+		return file{}, err
+	}
+
+	defer csvFile.Close()
+
+	return ParseOpenedCsvFile(csvFile, hasHeaders)
 }
 
-func ParseCsvJsonFile(f *os.File, hasHeaders bool) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+func ParseOpenedCsvFile(f *os.File, hasHeaders bool) (file, error) {
+	// parsing csv data
+	data, err := csv.NewReader(f).ReadAll()
+	if err != nil {
+		return file{}, err
+	}
```

Comment:
Read the file as before. i.e. with: `byteValue, err := ioutil.ReadAll(f)`
Then in `ParseCsv()`, you can do the following:

```golang
records := csv.NewReader(bytes.NewBuffer(c)).ReadAll()
```